### PR TITLE
ARROW-1015 [Java] Schema-level metadata

### PR DESCRIPTION
**Pull request by elahrvivaz**

### CODE REVIEW

File: java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Schema.java [hunk 1]
```diff
@@ -76,24 +84,43 @@ public static Schema convertSchema(org.apache.arrow.flatbuf.Schema schema) {
...
     this.fields = Collections.unmodifiableList(fieldList);
+    this.metadata = metadata == null ? ImmutableMap.<String, String>of() : ImmutableMap.copyOf(metadata);
   }
 
   public List<Field> getFields() {
     return fields;
   }
 
+  @JsonInclude(Include.NON_EMPTY)
+  public Map<String, String> getMetadata() {
```
Comment (julienledem) [inline java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Schema.java]: getCustomMetadata?
We should be consistent with FBS

**Review by julienledem — COMMENTED** [review]
One minor comment. Otherwise this looks good to me.

File: java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Schema.java (hunk 1 above)
Comment (elahrvivaz) [inline java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Schema.java]: fixed

**Review by julienledem — APPROVED** [review]
+1

**Comment by wesm** [comment]
Merging; the build failure was caused by PARQUET-987

**Pull request closed without merging**